polkadot/node/core/pvf/common/src/worker/mod.rs

```diff
@@ -184,6 +221,36 @@ macro_rules! decl_worker_main {
 	};
 }
 
+// os_pipe lib
+#[cfg(not(any(target_os = "ios", target_os = "macos", target_os = "haiku")))]
+pub fn pipe2_cloexec() -> io::Result<(libc::c_int, libc::c_int)> {
```

Comment:
Nice, just needs:

1. A better comment with a bit more context. Something like "taken from the os_pipe crate. Copied here to reduce one dependency and because its type-safe abstractions do not play well with `nix`'s `clone`.
2. Let's remove the `ios` and `haiku` targets since we don't care about those. 😁 We only target unix systems like mac and linux.